app/models/canonical_material.rb

```diff
@@ -1,6 +1,9 @@
 # frozen_string_literal: true
 
 class CanonicalMaterial < ApplicationRecord
+  has_many :smithable_armors, through: :canonical_armors_smithing_materials, source: :canonical_armor
+  has_many :temperable_armors, through: :canonical_armors_tempering_materials, source: :canonical_armor
+
   validates :name, presence: true, uniqueness: true
-  validates :unit_weight, numericality: { greater_than_or_equal_to: 0 }
+  validates :unit_weight, presence: true, numericality: { greater_than_or_equal_to: 0 }
```

Comment:
Again, ensures a clear message to users in the event they leave unit weight blank that the change they need to make is to add a unit weight.